tests/containers/account/visualizations/AccountRankVisualizationContainer-test.jsx

```diff
@@ -0,0 +1,227 @@
+                    });
+                }
+                else {
+                    reject({
+                        data: expectedResponse
+                    });
+                }
+            });
+        });
+
+        return {
+            promise: networkCall,
+            cancel: jest.fn()
+        };
+    });
+};
+
+const unmockAccountHelper = () => {
+    jest.useRealTimers();
+    jest.unmock('helpers/accountHelper');
+};
+
+const defaultFilters = {
+    dateType: 'fy',
+    fy: new Set(),
+    startDate: null,
+    endDate: null
+};
+
+describe('AccountRankVisualizationContainer', () => {
+    it('should load data on mount', () => {
+        mockAccountHelper('fetchTasCategoryTotals', 'resolve', mockCategories);
+
+        mount(<AccountRankVisualizationContainer
+            reduxFilters={defaultFilters}
+            account={mockReduxAccount} />);
+
+        jest.runAllTicks();
+
+        expect(fetchDataSpy.callCount).toEqual(1);
+        fetchDataSpy.reset();
+    });
+
+    it('should reload data when the filters change', () => {
+        mockAccountHelper('fetchTasCategoryTotals', 'resolve', mockCategories);
+
+        const container = mount(<AccountRankVisualizationContainer
+            reduxFilters={defaultFilters}
+            account={mockReduxAccount} />);
+
+        jest.runAllTicks();
+
+        expect(fetchDataSpy.callCount).toEqual(1);
+
+        container.setProps({
+            reduxFilters: Object.assign({}, defaultFilters, {
+                dateType: 'dr'
+            })
+        });
+
+        jest.runAllTicks();
+
+        expect(fetchDataSpy.callCount).toEqual(2);
+        fetchDataSpy.reset();
+    });
+
+    describe('parseData', () => {
+        it('should parse the API response and update the container state with series data', () => {
+            const container = shallow(<AccountRankVisualizationContainer
+                reduxFilters={defaultFilters}
+                account={mockReduxAccount} />);
+
+            container.instance().parseData(mockCategories);
+
+            const expectedState = {
+                labelSeries: ['709'],
+                dataSeries: [2696684.86],
+                descriptions: ['Obligated balance for 709: $2,696,685'],
+                loading: false,
+                total: 1,
+                page: 1,
+                categoryScope: 'program_activity'
+            };
+
+            expect(container.state()).toEqual(expectedState);
+        });
+    });
+
+    describe('changeScope', () => {
+        it('should reset the page number and update the state', () => {
+            const container = shallow(<AccountRankVisualizationContainer
+                reduxFilters={defaultFilters}
+                account={mockReduxAccount} />);
+
+            container.setState({
+                page: 5
+            });
+
+            expect(container.state().page).toEqual(5);
+
+            container.instance().changeScope('something else');
```

Comment:
No, because at time of development we don't know what the API values for other scopes would be (also the actual scope value is irrelevant to the logic of the test).